options.go

```diff
@@ -0,0 +1,453 @@
+	}
+	if cfg.FlowSimpleCollections != nil {
+		optList = append(optList, WithFlowSimpleCollections(*cfg.FlowSimpleCollections))
+	}
+	if cfg.KnownFields != nil {
+		optList = append(optList, WithKnownFields(*cfg.KnownFields))
+	}
+	if cfg.SingleDocument != nil && *cfg.SingleDocument {
+		optList = append(optList, WithSingleDocument())
+	}
+	if cfg.UniqueKeys != nil {
+		optList = append(optList, WithUniqueKeys(*cfg.UniqueKeys))
+	}
+	if cfg.Canonical != nil {
+		optList = append(optList, WithCanonical(*cfg.Canonical))
+	}
+	if cfg.LineBreak != nil {
+		switch *cfg.LineBreak {
+		case "ln":
+			optList = append(optList, WithLineBreak(LNBreak))
+		case "cr":
+			optList = append(optList, WithLineBreak(CRBreak))
+		case "crln":
+			optList = append(optList, WithLineBreak(CRLNBreak))
+		default:
+			return nil, errors.New("yaml: invalid line-break value (use ln, cr, or crln)")
+		}
+	}
+
+	return Options(optList...), nil
+}
+
+// WithPlugins adds multiple plugins at once.
+//
+// Example:
+//
+//	yaml.Dump(&data, yaml.V4, yaml.WithPlugins(comments.V4()))
+func WithPlugins(plugins ...Plugin) Option {
+	return func(o *options) error {
+		for _, p := range plugins {
+			if p == nil {
+				return errors.New("yaml: plugin cannot be nil")
+			}
+			o.plugins = append(o.plugins, p)
+		}
+		return nil
+	}
+}
+
+// V2 provides go-yaml v2 formatting defaults:
+//   - 2-space indentation
+//   - Non-compact sequence indentation
+//   - 80-character line width
+//   - Unicode enabled
+//   - Unique keys enforced
+//
+// Usage:
+//
+//	yaml.Dump(&data, yaml.V2)
+//	yaml.Dump(&data, yaml.V2, yaml.WithIndent(4))
+var V2 = Options(
+	WithIndent(2),
+	WithLineWidth(80),
+	WithUnicode(true),
+	WithUniqueKeys(true),
+)
+
+// V3 provides go-yaml v3 formatting defaults:
+//   - 4-space indentation (classic go-yaml v3 style)
+//   - Non-compact sequence indentation
+//   - 80-character line width
+//   - Unicode enabled
+//   - Unique keys enforced
+//
+// Usage:
+//
+//	yaml.Dump(&data, yaml.V3)
+//	yaml.Dump(&data, yaml.V3, yaml.WithIndent(2))
+var V3 = Options(
+	WithIndent(4),
+	WithLineWidth(80),
+	WithUnicode(true),
+	WithUniqueKeys(true),
+)
+
+// V4 provides go-yaml v4 formatting defaults:
+//   - 2-space indentation (more compact than v3)
+//   - Compact sequence indentation
+//   - 80-character line width
+//   - Unicode enabled
+//   - Unique keys enforced
+//
+// Usage:
+//
+//	yaml.Dump(&data, yaml.V4)
+//	yaml.Dump(&data, yaml.V4, yaml.WithPlugins(comments.V4()))
+var V4 = Options(
+	WithIndent(2),
+	WithCompactSeqIndent(true),
+	WithLineWidth(80),
+	WithUnicode(true),
+	WithUniqueKeys(true),
+)
+
+// applyOptions applies the given options to a new options struct.
+// Starts with v4 defaults.
+func applyOptions(opts ...Option) (*options, error) {
+	o := &options{
+		// v4 defaults
+		indent:           2,
+		compactSeqIndent: true,
+		lineWidth:        80,
+		unicode:          true,
+		uniqueKeys:       true,
```

Comment:
Another way to consider the v4 default would be to prepend V4 to the opts list.

This way the default are really the v4 ones